src/test/java/org/graylog/integrations/notifications/types/SlackEventNotificationTest.java

```diff
@@ -0,0 +1,154 @@
+package org.graylog.integrations.notifications.types;
+
+import com.floreysoft.jmte.Engine;
+import com.google.common.collect.ImmutableList;
+import org.bson.types.ObjectId;
+import org.graylog.events.notifications.*;
+import org.graylog.events.notifications.types.HTTPEventNotificationConfig;
+import org.graylog.integrations.notifications.modeldata.StreamModelData;
+import org.graylog2.notifications.Notification;
+import org.graylog2.notifications.NotificationImpl;
+import org.graylog2.notifications.NotificationService;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.MessageSummary;
+import org.graylog2.plugin.streams.Stream;
+import org.graylog2.plugin.system.NodeId;
+import org.graylog2.shared.bindings.providers.ObjectMapperProvider;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SlackEventNotificationTest extends SlackPluginTestFixture {
```

Comment:
the rest of the event notifications do not follow the factory pattern. we can explore this in the future. what do you think @waab76

Reply:
None of the other event notifications need to get a new client object every time the notification config changes.  If they did, it would make sense to use a factory pattern for getting new client objects rather than calling the constructor directly.

I feel like this is worth addressing as part of this PR as it will let us have cleaner and clearer tests.  The `SlackEventNotification` code has no reason to know or care about the existence of OKHttp and yet, because of the way it is constructing `SlackClient` objects, if something related to OKHttp breaks, then the tests for `SlackEventNotification` will break as well.  The JUnit tests should only break when the class itself breaks, not when other classes break.